File: src/arrays/duplicateinteger/DuplicateInteger.java
package arrays.duplicateinteger;


import java.util.HashSet;
import java.util.Set;

/*
   @author Austin Oyugi
   @since 22/07/2024
   @mail austinoyugi@gmail.com
*/
public class DuplicateInteger {

    public static boolean hasDuplicates(int[] nums) {

        // Set is a data structure that does not allow duplicates
        // When you give it a duplicate value only the previous reference will remain
        Set<Integer> numsSet = new HashSet<>();

        // loop and add the values to the set
        for (int num : nums) {
            numsSet.add(num);
        }

        // If the length is not equal then a duplicate must have been found
        return nums.length != numsSet.size();
    }

    public static void main(String[] args) {

        int[] test1 = {1, 2, 3, 4, 5};

        // Expect false
        System.out.println("Has duplicates " + hasDuplicates(test1));
    }
}

File: src/arrays/duplicateinteger/duplicate_integer.rs
use std::collections::HashSet;

fn has_duplicates(arr : &[u8]) -> bool {

    // Set is a data structure that does not allow duplicates
    // When you give it a duplicate value only the previous reference will remain
    let mut set = HashSet::new();

    // loop and add the values to the set
    for number in arr {
        set.insert(number);
    }

    // If the lengths are not equal then a duplicate must have been found
    arr.len() != set.len()
}

pub fn main(){
    // Expect false
    println!("Has duplicates {:?}",has_duplicates(&[1,2,3,4,5]));
}
